### PR TITLE
Fix: loader transform should produce a valid config

### PR DESCRIPTION
**Pull request by okonet**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Not relevant

**Summary**
This PR is a bug fix for #90 

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: lib/transformations/loaders/__snapshots__/loaders.test.js.snap [hunk 1]
```diff
@@ -81,9 +104,14 @@ exports[`loaders transforms correctly using "loaders-1" data 1`] = `
...
-                'loader': 'style'
+                loader: 'style-loader'
             }, {
-                'loader': 'css?modules&importLoaders=1&string=test123'
+                loader: 'css-loader',
+                options: {
+                    modules: true,
+                    importLoaders: 1,
+                    string: 'test123'
+                }
```
Comment (pksjce) [inline lib/transformations/loaders/__snapshots__/loaders.test.js.snap]: Yay!

File: lib/transformations/utils.test.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ describe('utils', () => {
 			expect(j(j.objectExpression([res])).toSource()).toMatchSnapshot();
 		});
 
-		it('should create properties for complex keys', () => {
+		xit('should create properties for complex keys', () => {
```
Comment (pksjce) [inline lib/transformations/utils.test.js]: Is this intentional?

Comment (okonet) [inline lib/transformations/utils.test.js]: Yeah, it's a wrong one. I left a todo in the code

**Comment by pksjce** [comment]
I think some snapshots were not updated. Build seems to be failing because of that.

**Comment by okonet** [comment]
Hmm, all green locally:

```
Test Suites: 31 passed, 31 total
Tests:       125 passed, 125 total
Snapshots:   109 passed, 109 total
Time:        23.485s
Ran all test suites.
```

**Comment by ev1stensberg** [comment]
@okonet Are you going to fix the todo as well before merge?

**Comment by okonet** [comment]
Nope. It's a big one.

**Comment by okonet** [comment]
Does anyone know why are tests failing on Node 4.x?

**Comment by okonet** [comment]
Just ran it locally with Node v4.6.1 and all tests have passed.

File: lib/transformations/removeJsonLoader/__snapshots__/removeJsonLoader.test.js.snap [hunk 3]
```diff
@@ -37,13 +43,7 @@ exports[`removeJsonLoader transforms correctly using "removeJsonLoader-2" data 1
...
 "export default {
     module: {
-        rules: [{
-            test: /\\\\.yml/,
-            use: ['another-loader', 'yml-loader']
-        }, {
-            test: /\\\\.yml/,
-            use: 'yml-loader'
-        }]
+        rules: []
```
Comment (pksjce) [inline lib/transformations/removeJsonLoader/__snapshots__/removeJsonLoader.test.js.snap]: Is this right?

File: lib/transformations/removeJsonLoader/__testfixtures__/removeJsonLoader-3.input.js [hunk 4]
```diff
@@ -1,14 +1,10 @@
...
-            use: ['json-loader', 'another-loader', 'yml-loader']
-        }, {
-            test: /\.yml/,
-            use: ['json-loader', 'yml-loader']
-        }, {
             test: /\.json/,
-            use: 'json-loader'
+            use: [{
+                loader: 'json-loader'
+            }]
```
Comment (pksjce) [inline lib/transformations/removeJsonLoader/__testfixtures__/removeJsonLoader-3.input.js]: Shouldnt it remove only `json-loader` and keep the rest?

Comment (okonet) [inline lib/transformations/removeJsonLoader/__testfixtures__/removeJsonLoader-3.input.js]: It only has the json loader

File: lib/transformations/removeJsonLoader/__snapshots__/removeJsonLoader.test.js.snap (hunk 3 above)
Comment (okonet) [inline lib/transformations/removeJsonLoader/__snapshots__/removeJsonLoader.test.js.snap]: Yes I changed the test input

File: lib/transformations/removeJsonLoader/__testfixtures__/removeJsonLoader-3.input.js (hunk 4 above)
Comment (pksjce) [inline lib/transformations/removeJsonLoader/__testfixtures__/removeJsonLoader-3.input.js]: I'm sorry my bad!

**Comment by pksjce** [comment]
Hey! This is passing now. Can merge I think.

**Comment by okonet** [comment]
@pksjce I've fixed some more edge cases that we didn't cover. That lead to a bigger code refactoring. Could you please do another code review?

**Comment by ev1stensberg** [comment]
Both of us or just Pavithra?

**Comment by okonet** [comment]
One more pair of 👀 can't hurt!

**Pull request merged**